tests/test_cli.py

```diff
@@ -138,13 +157,21 @@ def should_report_dataset_creation_success(
     response_payload_object = get_response_object(
         HTTPStatus.CREATED, {DATASET_ID_SHORT_KEY: dataset_id}
     )
-    response_payload = json_dict_to_file_object(response_payload_object)
+    response_payload_encode = dumps(response_payload_object).encode()
+    response_payload = stream_file_like_object(response_payload_encode)
     boto3_client_mock.return_value.invoke.return_value = InvocationResponseTypeDef(
         StatusCode=HTTPStatus.OK,
         FunctionError="",
         LogResult="",
         Payload=response_payload,
         ExecutedVersion=LAMBDA_EXECUTED_VERSION,
+        ResponseMetadata={
+            "RequestId": any_request_id(),
+            "HostId": any_host(),
+            "HTTPStatusCode": any_http_status_code(),
+            "HTTPHeaders": {},
+            "RetryAttempts": any_retry_attempts(),
+        },
```

Comment:
Since this `dict` is *only* random data we could probably just pull it out as an `any_response_metadata`. What do you think? By the way, if you use the `ResponseMetadataTypeDef` constructor instead of a plain `dict` we can get more info about the structure as it evolves.